lib/ansible/modules/windows/win_domain.ps1

```diff
@@ -127,6 +127,9 @@ if (-not $forest) {
         if ($_.Exception.ExitCode -eq 15) {
             $result.reboot_required = $true
         } else {
+            if ($_.Exception.Message.Contains("A reboot is required.")) {
```

Comment:
Using the exception message to detect this is probably the wrong way to go, it would most likely fall flat as soon as another language is being used. Looking at https://docs.microsoft.com/en-us/windows-server/identity/ad-ds/deploy/troubleshooting-domain-controller-deployment it seems like we should include 19 as an ExitCode in the check above, e.g. `$_.Exception.ExitCode -in @(15, 19)`.

It might be a good idea ot include the ExitCode in the Fail-Json message as well to help detect these problems in the future.

Reply:
I encountered that issue on a german server, so I just assumed it is always english, but you're right we should not rely on the message and use exit codes instead.

Reply:
I don't see any try catch around the `Install-ADDSDomainController` except for the generic one. As I don't know if the error codes are equivalent, I'm not including it for now into this pr.

Reply:
> I don't see any try catch around the Install-ADDSDomainController except for the generic one

Hmm I must be imagining things then sorry.